### PR TITLE
Add node-startup-prep util

### PR DESCRIPTION
**Pull request by endersonmaia**
This PR will add a little utility to handle the issues reported at #428 e #403.

I'm forking at sunodo because I may need to release this binary there to use it soon, before I wait a possible integration into the cartesi/rollups-node code somehow.

Reviews are appreciated!

### CODE REVIEW

File: utils/node-startup-prep/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM golang:alpine AS builder
```
Comment (tuler) [inline utils/node-startup-prep/Dockerfile:1]: alpine? If you will use on debian you better use debian, even if the binary is statically compiled.

File: utils/node-startup-prep/main.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	if err != nil {
+		log.Fatalln("ERR: ", err)
+	}
+	var redisEndpoint = fmt.Sprintf("%s:%s", u.Hostname(), u.Port())
+
+	postgresEndpoint, exists := os.LookupEnv("CARTESI_POSTGRES_ENDPOINT")
+	if !exists {
+		log.Fatalln("ERR: Missing environment variable CARTESI_POSTGRES_ENDPOINT service=node-startup-prep")
+	}
+	log.Printf("CARTESI_POSTGRES_ENDPOINT=%s service=node-startup-prep", postgresEndpoint)
```
Comment (tuler) [inline utils/node-startup-prep/main.go:46]: Don't print database password

File: utils/node-startup-prep/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	_redisEndpoint, exists := os.LookupEnv("CARTESI_EXPERIMENTAL_SUNODO_VALIDATOR_REDIS_ENDPOINT")
+	if !exists {
+		log.Fatalln("ERR: Missing environment variable CARTESI_EXPERIMENTAL_SUNODO_VALIDATOR_REDIS_ENDPOINT")
+	}
+	log.Printf("CARTESI_EXPERIMENTAL_SUNODO_VALIDATOR_REDIS_ENDPOINT=%s service=node-startup-prep", _redisEndpoint)
+	u, err := url.Parse(_redisEndpoint)
+	if err != nil {
+		log.Fatalln("ERR: ", err)
+	}
+	var redisEndpoint = fmt.Sprintf("%s:%s", u.Hostname(), u.Port())
```
Comment (tuler) [inline utils/node-startup-prep/main.go:40]: Don't print redis password (if there is one)

File: utils/node-startup-prep/main.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	postgresEndpoint, exists := os.LookupEnv("CARTESI_POSTGRES_ENDPOINT")
+	if !exists {
+		log.Fatalln("ERR: Missing environment variable CARTESI_POSTGRES_ENDPOINT service=node-startup-prep")
+	}
+	log.Printf("CARTESI_POSTGRES_ENDPOINT=%s service=node-startup-prep", postgresEndpoint)
+
+	// Clean inputs and outputs streams
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     redisEndpoint,
+		Password: "", // no password set
```
Comment (tuler) [inline utils/node-startup-prep/main.go:51]: never has password?

**Review by tuler — COMMENTED** [review]
I don't like the name much.
Suggestion: `node-pre-start`

File: utils/node-startup-prep/Dockerfile (hunk 1 above)
Comment (endersonmaia) [reply utils/node-startup-prep/Dockerfile:1]: I think golang crosscompiles and has no links to any C library, so no worries here.

File: utils/node-startup-prep/main.go (hunk 4 above)
Comment (endersonmaia) [reply utils/node-startup-prep/main.go:51]: will fix it

File: utils/node-pre-start/main.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+	sunodoValidatorEnabled bool
+	redisEndpoint          *url.URL
+}
+
+func createPostgresDB(cfg nodePreStartConfig) error {
+	err := retry.Do(
+		func() error {
+			db, err := sqlx.Connect("postgres", cfg.postgresEndpoint.String())
+			if err != nil {
+				log.Println("ERR: POSTGRESQL: can't connect: ", err, "")
```
Comment (tuler) [inline utils/node-pre-start/main.go:34]: are you sure printing the err doesn't leak the password?

**Comment by endersonmaia** [comment]
I'm already maintaining an internal version of such tool, will wait for the Node Team to have hopefully something integrated on a future version of the multi-dapp node.

**Pull request closed without merging**